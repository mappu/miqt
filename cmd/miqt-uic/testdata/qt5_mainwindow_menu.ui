<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>29</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="separator"/>
    <addaction name="actionE_xit"/>
   </widget>
   <addaction name="menu_File"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionNew">
   <property name="text">
    <string>New...</string>
   </property>
   <property name="toolTip">
    <string>Custom Tooltip for New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionE_xit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
