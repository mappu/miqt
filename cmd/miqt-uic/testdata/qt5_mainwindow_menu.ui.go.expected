// Generated by miqt-uic. To update this file, edit the .ui file in
// Qt Designer, and then run 'go generate'.
//
//go:generate miqt-uic go_generate_args

package test_package_name

import (
	"github.com/mappu/miqt/qt"
)

type MainWindowUi struct {
	MainWindow    *qt.QMainWindow
	centralwidget *qt.QWidget
	menubar       *qt.QMenuBar
	menu_File     *qt.QMenu
	statusbar     *qt.QStatusBar
	actionNew     *qt.QAction
	actionE_xit   *qt.QAction
}

// NewMainWindowUi creates all Qt widget classes for MainWindow.
func NewMainWindowUi() *MainWindowUi {
	ui := &MainWindowUi{}
	ui.MainWindow = qt.NewQMainWindow(nil)
	ui.MainWindow.SetObjectName("MainWindow")
	ui.MainWindow.Resize(800, 600)
	ui.actionNew = qt.NewQAction()
	ui.actionNew.SetObjectName("actionNew")
	ui.actionE_xit = qt.NewQAction()
	ui.actionE_xit.SetObjectName("actionE_xit")
	ui.centralwidget = qt.NewQWidget(ui.MainWindow.QWidget)
	ui.centralwidget.SetObjectName("centralwidget")
	ui.MainWindow.SetCentralWidget(ui.centralwidget) // Set central widget
	ui.menubar = qt.NewQMenuBar(ui.MainWindow.QWidget)
	ui.menubar.SetObjectName("menubar")
	ui.menubar.Resize(800, 29)
	ui.menu_File = qt.NewQMenu(ui.menubar.QWidget)
	ui.menu_File.SetObjectName("menu_File")
	ui.menu_File.QWidget.AddAction(ui.actionNew)
	ui.menu_File.AddSeparator()
	ui.menu_File.QWidget.AddAction(ui.actionE_xit)
	ui.menubar.AddMenu(ui.menu_File)
	ui.MainWindow.SetMenuBar(ui.menubar)
	ui.statusbar = qt.NewQStatusBar(ui.MainWindow.QWidget)
	ui.statusbar.SetObjectName("statusbar")
	ui.MainWindow.SetStatusBar(ui.statusbar)

	ui.Retranslate()

	return ui
}

// Retranslate reapplies all text translations.
func (ui *MainWindowUi) Retranslate() {
	ui.MainWindow.SetWindowTitle(qt.QCoreApplication_Tr("MainWindow"))
	ui.actionNew.SetText(qt.QMainWindow_Tr("New..."))
	ui.actionNew.SetToolTip(qt.QMainWindow_Tr("Custom Tooltip for New"))
	ui.actionNew.SetShortcut(qt.NewQKeySequence2(qt.QMainWindow_Tr("Ctrl+N")))
	ui.actionE_xit.SetText(qt.QMainWindow_Tr("E&xit"))
	ui.menu_File.SetTitle(qt.QMenuBar_Tr("&File"))
}
